### PR TITLE
js: Move everything to pnpm

### PR DESCRIPTION
**Pull request by joncinque**
#### Problem

There's a lot of libraries in SPL, and managing all of their dependencies can take awhile, especially as all of the dependabot PRs come in.

#### Solution

Move everything to one pnpm-lock.yaml file! As I was going through the different packages, there were loads of issues, so I had to fix up a few other things. This is step 1, and after this, we can change all of the deps in the package.json files to use the version from `workspace:*`

Note: this is currently set to draft while CI gets happy

### CODE REVIEW

File: .github/workflows/pull-request-stake-pool.yml [hunk 1]
```diff
@@ -99,11 +98,6 @@ jobs:
           key: node-${{ hashFiles('pnpm-lock.yaml') }}
           restore-keys: |
             node-
-      - name: Download programs
-        uses: actions/download-artifact@v2
-        with:
-          name: stake-pool-programs
-          path: target/deploy
```
Comment (buffalojoec) [inline .github/workflows/pull-request-stake-pool.yml:103]: How come we don't need to download the programs anymore?

File: token-swap/js/package.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
   "scripts": {
     "build": "tsc -p tsconfig.json && tsc-esm -p tsconfig.json && tsc -p tsconfig.cjs.json",
     "postbuild": "echo '{\"type\":\"commonjs\"}' > dist/cjs/package.json && echo '{\"type\":\"module\"}' > dist/esm/package.json",
-    "test": "ts-node test/main.ts",
+    "test": "mocha test",
```
Comment (buffalojoec) [inline token-swap/js/package.json:36]: This is fine, but I'm curious if we want align on either `jest` or `mocha` for all JS libs?

File: token-swap/js/.eslintrc
Comment (buffalojoec) [inline token-swap/js/.eslintrc:1]: We _could_ enforce `eslint` and `prettier` at the root level for all libs (like `rustfmt`). `pnpm` will pick it up.

Also, have you seen `@solana/eslint-config-solana`?

**Review by buffalojoec — COMMENTED** [review]
I love it! This should make managing JS libraries in SPL much easier.

File: stake-pool/js/package.json [hunk 3]
```diff
@@ -49,7 +49,8 @@
     "@solana/spl-token": "^0.3.7",
     "@solana/web3.js": "^1.76.0",
     "bn.js": "^5.2.0",
-    "buffer": "^6.0.3"
+    "buffer": "^6.0.3",
+    "superstruct": "^0.14.2"
```
[the remark this answers is not in the document]
Comment (joncinque) [reply stake-pool/js/package.json:53]: Yeah, I guess it was picking it up through a transitive dependency, but the older build tools didn't care?

File: .github/workflows/pull-request-stake-pool.yml (hunk 1 above)
Comment (joncinque) [reply .github/workflows/pull-request-stake-pool.yml:103]: The JS tests don't actually use the program, so I decided to simplify it for faster runs

File: token-swap/js/package.json (hunk 2 above)
Comment (joncinque) [reply token-swap/js/package.json:36]: We probably want to use `jest`, but it was such a nightmare to get right in the memo directory I decided to keep it simple here. We're probably dropping all of this stuff soon anyway

File: token-swap/js/.eslintrc
Comment (joncinque) [reply token-swap/js/.eslintrc:1]: Ah cool, I added that as an item in #5785, and I can definitely use the `@solana` version!

File: token-swap/js/package.json (hunk 2 above)
Comment (buffalojoec) [reply token-swap/js/package.json:36]: You mean the example programs?

Comment (joncinque) [reply token-swap/js/package.json:36]: yeah, a lot of the unmaintained reference implementations

**Review by buffalojoec — APPROVED** [review]

**Pull request merged**